lib/exgboost/parameters.ex

```diff
@@ -0,0 +1,968 @@
+      type: {:custom, EXGBoost.Parameters, :in_range, ["(0,1]"]},
+      default: 1.0,
+      doc: """
+      Subsample ratio of the training instance. Setting it to `0.5` means that XGBoost
+      randomly collected half of the data instances to grow trees and this will prevent
+      overfitting. Subsampling will occur once in every boosting iteration. Valid range is `(0, 1]`.
+      """
+    ],
+    sampling_method: [
+      type: {:in, [:uniform, :gradient_based]},
+      default: :uniform,
+      doc: """
+      The method to use to sample the training instances.
+          * `:uniform`: each training instance has an equal probability of being selected.
+            Typically set subsample >= 0.5 for good results.
+          * `:gradient_based`: the selection probability for each training instance is proportional
+              to the regularized absolute value of gradients. subsample may be set to as low as 0.1
+              without loss of model accuracy. Note that this sampling method is only supported when
+              `tree_method` is set to `gpu_hist`; other tree methods only support `:uniform` sampling.
+      """
+    ],
+    colsample_by: [
+      keys: [
+        tree: [
+          type: {:custom, EXGBoost.Parameters, :in_range, ["(0,1]"]},
+          default: 1,
+          doc: """
+          The subsample ratio of columns when constructing each tree. Subsampling occurs once for every tree constructed. Valid range is `(0, 1]`.
+          """
+        ],
+        level: [
+          type: {:custom, EXGBoost.Parameters, :in_range, ["(0,1]"]},
+          default: 1,
+          doc: """
+          The subsample ratio of columns for each level. Subsampling occurs once for every new depth level reached in a tree. Columns are subsampled from the set of columns chosen for the current tree. Valid range is `(0, 1]`.
+          """
+        ],
+        node: [
+          type: {:custom, EXGBoost.Parameters, :in_range, ["(0,1]"]},
+          default: 1,
+          doc: """
+          The subsample ratio of columns for each node (split). Subsampling occurs once every time a new split is evaluated. Columns are subsampled from the set of columns chosen for the current level. Valid range is `(0, 1]`.
+          """
+        ]
+      ],
+      type: :keyword_list,
+      doc: """
+      This is a family of parameters for subsampling of columns.
+      All `colsample_by*` parameters have a range of `(0, 1]`, the default value of `1`, and specify the fraction of columns to be subsampled.
+      `colsample_by*` parameters work cumulatively. For instance, the combination
+      `col_sampleby: [tree: 0.5, level: 0.5, node: 0.5]` with `64` features will leave `8`
+      """
+    ],
+    lambda: [
+      type: {:custom, EXGBoost.Parameters, :in_range, ["[0,:inf]"]},
+      default: 1,
+      doc: """
+      L2 regularization term on weights. Increasing this value will make model more conservative.
+      Valid range is `[0, :infinity]`.
```

Comment:
Just a reminder to choose between latex, elixir code, atoms

Reply:
I think I'll do latex, definitely need to normalize that notation